tests/src/test/java/alluxio/testutils/IntegrationTestUtils.java

```diff
@@ -117,18 +195,24 @@ public static void waitForBlocksToBeFreed(final BlockWorker bw, final Long... bl
       throws TimeoutException {
     try {
       // Execute 1st heartbeat from worker.
-      HeartbeatScheduler.execute(HeartbeatContext.WORKER_BLOCK_SYNC);
+      Whitebox.getInternalState(bw, BlockMasterSync.class).heartbeat();
 
       // Waiting for the blocks to be added into the heartbeat reportor, so that they will be
       // removed from master in the next heartbeat.
       CommonUtils.waitFor("blocks to be removed", () -> {
-        BlockHeartbeatReporter reporter = Whitebox.getInternalState(bw, "mHeartbeatReporter");
-        List<Long> blocksToRemove = Whitebox.getInternalState(reporter, "mRemovedBlocks");
-        return blocksToRemove.containsAll(Arrays.asList(blockIds));
+        if (blockIds.length == 0) {
+          return true;
+        }
+        Set<Long> blocks = bw.getBlockStore().getBlockStoreMetaFull()
+            .getBlockListByStorageLocation().values().stream().flatMap(List::stream)
+            .collect(Collectors.toSet());
+        List<Long> bids = Lists.newArrayList(blockIds);
+        return bids.stream().map(id -> !blocks.contains(id)).reduce((s1, s2) -> s1 && s2)
```

Comment:
can we not do a set intersection instead of this?

Reply:
good call